src/SmartComponents/CreateImageWizard/CreateImageWizard.js

```diff
@@ -131,38 +191,51 @@ class CreateImageWizard extends Component {
     }
 
     onSave () {
-        let request = {
-            distribution: this.state.release,
-            image_requests: [
-                {
-                    architecture: 'x86_64',
-                    image_type: 'ami',
-                    upload_requests: [{
-                        type: 'aws',
-                        options: {
-                            share_with_accounts: this.state.upload.options.share_with_accounts,
+        let requests = [];
+        Object.keys(this.state.uploadDestinations).forEach(provider => {
+            switch (provider) {
+                case 'aws': {
+                    let request = {
+                        distribution: this.state.release,
+                        image_requests: [
+                            {
+                                architecture: this.state.arch,
+                                image_type: 'ami',
+                                upload_requests: [ this.state.uploadAWS ],
+                            }],
+                        customizations: {
+                            subscription: this.state.subscription,
                         },
-                    }],
-                }],
-            customizations: {
-                subscription: this.state.subscription,
-            },
-        };
+                    };
+                    requests.push(request);
+                    break;
+                }
+
+                case 'azure':
+                    break;
+                case 'google':
+                    break;
+                default:
+                    break;
+            }
+        });
 
-        let { updateCompose } = this.props;
-        api.composeImage(request).then(response => {
-            let compose = {};
-            compose[response.id] = {
-                image_status: {
-                    status: 'pending',
-                },
-                distribution: request.distribution,
-                architecture: request.image_requests[0].architecture,
-                image_type: request.image_requests[0].image_type,
-            };
-            updateCompose(compose);
-            this.props.history.push('/landing');
+        requests.forEach(request => {
+            api.composeImage(request).then(response => {
+                let compose = {};
+                compose[response.id] = {
+                    image_status: {
+                        status: 'pending',
+                    },                    distribution: request.distribution,
+                    architecture: request.image_requests[0].architecture,
+                    image_type: request.image_requests[0].image_type,
+                };
+                this.props.updateCompose(compose);
+                this.props.history.push('/landing');
+            });
         });
+
+        // Promise.all(composeRequests).then(() => this.props.history.push('/landing'));
```

Comment:
Might as well implement this already no?

Reply:
Oh yes. I totally missed this.